ui/pages/institutional/custody/custody.js

```diff
@@ -0,0 +1,677 @@
+          flexDirection={FLEX_DIRECTION.ROW}
+          justifyContent={JustifyContent.spaceBetween}
+          alignItems={AlignItems.center}
+          borderColor={BorderColor.borderDefault}
+          borderRadius={BorderRadius.SM}
+          padding={[3, 4]}
+          marginBottom={4}
+        >
+          <Box display={DISPLAY.FLEX} alignItems={AlignItems.center}>
+            {custodian.iconUrl && (
+              <img
+                width={32}
+                height={32}
+                src={custodian.iconUrl}
+                alt={custodian.displayName}
+              />
+            )}
+            <Text marginLeft={2}>{custodian.displayName}</Text>
+          </Box>
+
+          <Button
+            size={BUTTON_SIZES.SM}
+            data-testid="custody-connect-button"
+            onClick={async (_) => {
+              const jwtListValue = await dispatch(
+                mmiActions.getCustodianJWTList(custodian.name),
+              );
+              setSelectedCustodianName(custodian.name);
+              setSelectedCustodianType(custodian.type);
+              setSelectedCustodianImage(custodian.iconUrl);
+              setSelectedCustodianDisplayName(custodian.displayName);
+              setApiUrl(custodian.apiUrl);
+              setCurrentJwt(jwtListValue[0] || '');
+              setJwtList(jwtListValue);
+              trackEvent({
+                category: 'MMI',
+                event: 'Custodian Selected',
+                properties: {
+                  custodian: custodian.name,
+                },
+              });
+            }}
+          >
+            {t('select')}
+          </Button>
+        </Box>,
+      );
+    });
+
+    return (
+      <Box
+        padding={[0, 7, 2]}
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.COLUMN}
+      >
+        <ButtonIcon
+          ariaLabel={t('back')}
+          iconName={IconName.ArrowLeft}
+          size={IconSize.Sm}
+          color={Color.iconDefault}
+          onClick={() => history.push(DEFAULT_ROUTE)}
+          display={DISPLAY.FLEX}
+        />
+        <Text
+          as="h4"
+          variant={TextVariant.bodyLgMedium}
+          marginTop={4}
+          marginBottom={4}
+        >
+          {t('connectCustodialAccountTitle')}
+        </Text>
+        <Text
+          as="h6"
+          color={TextColor.textDefault}
+          marginTop={2}
+          marginBottom={5}
+        >
+          {t('connectCustodialAccountMsg')}
+        </Text>
+        <Box>
+          <ul width={BLOCK_SIZES.FULL}>{custodianButtons}</ul>
+        </Box>
+      </Box>
+    );
+  };
+
+  const cancelConnectCustodianToken = () => {
+    setSelectedCustodianName('');
+    setSelectedCustodianType('');
+    setSelectedCustodianImage(null);
+    setSelectedCustodianDisplayName('');
+    setApiUrl('');
+    setCurrentJwt('');
+    setConnectError('');
+    setSelectError('');
+  };
+
+  const renderConnectButton = () => {
```

Comment:
[REQ] Please remove all these renderMethods. Just render them in the JSX that is returned at the component level. As some advice, if you find yourself reaching for abstracting parts of your tree into separate functions you are likely either over thinking the complexity of the JSX or the component you are making is too large and should be further broken into sub pieces that are their own files and own components.

Reply:
Will do 👍🏼